File: pocketc_backend/src/main/java/com/ssafy/pocketc_backend/domain/mission/entity/Mission.java
package com.ssafy.pocketc_backend.domain.mission.entity;

import com.ssafy.pocketc_backend.domain.event.entity.Status;
import com.ssafy.pocketc_backend.domain.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "missions")
@ToString(exclude = "user")
public class Mission {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer missionId;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private Integer subId;

    @Column(length = 512)
    private String dsl;

    @Column(length = 512, nullable = false)
    private String mission;

    private Integer type;

    private LocalDateTime validFrom;
    private LocalDateTime validTo;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private Status status = Status.IN_PROGRESS;

    private Integer progress;
}

File: pocketc_ai/app/schemas/mission.py
from datetime import date, datetime
from typing import List

from pydantic import BaseModel


class Mission(BaseModel):
    missionId: int
    mission: str
    subId: int       # 소분류 카테고리 보내기
    missionDsl: str  # 미션 해결을 위한 DSL
    missionType: int  # 데일리 0, 주간 1, 월간 2
    validFrom: datetime
    validTo: datetime

class Missions(BaseModel):
    userId : int
    date: date
    missions: List[Mission]

class MissionResponse(BaseModel):
    status: int
    message: str
    data: Missions
